omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,341 @@
+	}
+
+	/**
+	 *  Delete all Allergies
+	 */
+	@Test
+	public void shouldDeleteAllAllergies() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		Allergies allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertEquals(Allergies.SEE_LIST, allergies.getAllergyStatus());
+		Assert.assertFalse(allergy.isVoided());
+
+		// attempt to delete all allergies
+		handle(newDeleteRequest(getURI(), new Parameter("reason", "unit test")));
+		
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+	}
+
+	/*
+	 *  Get Allergy 
+	 */
+	@Test
+	public void shouldGetAllergyByUuid() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+
+		// attempt to get allergy by uuid
+		SimpleObject savedAllergy = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+
+		Assert.assertEquals(allergy.getComment(), Util.getByPath(savedAllergy, "comment"));
+		Assert.assertEquals(allergy.getSeverity().getUuid(), Util.getByPath(savedAllergy, "severity/uuid"));
+
+		if (allergy.getAllergen().isCoded()) {
+			Assert.assertEquals(allergy.getAllergen().getCodedAllergen().getUuid(), Util.getByPath(savedAllergy, "allergen/codedAllergen/uuid"));
+		} else {
+			Assert.assertEquals(allergy.getAllergen().getCodedAllergen().getUuid(), Util.getByPath(savedAllergy, "allergen/codedAllergen/uuid"));
+			Assert.assertEquals(allergy.getAllergen(), Util.getByPath(savedAllergy, "allergen/nonCodedAllergen"));
+		}
+		Assert.assertEquals(allergy.getAllergen().getAllergenType().toString(), Util.getByPath(savedAllergy, "allergen/allergenType"));
+	}
+
+	/**
+	 *  Save Allergy with Coded Allergen
+	 */
+	@Test
+	public void shouldSaveAllergyWithCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 4);
+		
+		// save allergy with coded allergen
+		String json = "{" +
+				        " \"comment\" : \"allergy comment\"," +
+				        " \"severity\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\"}," +
+				        " \"allergen\" : " + 
+				          " { \"allergenType\" : \"DRUG\", " +
+				          " \"codedAllergen\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\"} " +
+				          " } " +
+		              "}";
+		
+		// save allergy 
+		SimpleObject savedAllergy = deserialize(handle(newPostRequest(getURI(), json)));
+
+		Assert.assertEquals("allergy comment", Util.getByPath(savedAllergy, "comment"));

```

Comment:
How about here? Can't we just fetch allergies after posting and assert on allergy properties?